### PR TITLE
AL-259: Prove multi-designation financial truth acceptance

### LINKED ISSUE #259: Multi-Designation Financial Truth
## What to build

Replace single-designation assumptions with a first-class designation set in shared row and detail payloads. Every designation line is equal, fund-backed, and reconciles to the effective gift amount.

## Acceptance criteria

- [ ] Detail payload exposes a designation set, not a single primary designation object.
- [ ] Every designation line includes amount, currency, fund, fund type, and relevant memo/restriction context where available.
- [ ] The designation set reconciles to the effective gift amount.
- [ ] No designation is labeled or treated as primary in contribution detail.
- [ ] CRM and Hub summaries derive from the same designation set.
- [ ] Tests cover split gifts, General Fund fallback, and invalid fundless designation states.

## Blocked by

- https://github.com/Asymmetric-al/core/issues/256
- https://github.com/Asymmetric-al/core/issues/257

### PR DESCRIPTION
**Pull request by cobmojo**
## Summary

- Issue #259's implementation shipped with the contribution-shared split PRs (ADR-CD-008/009/010): the detail payload exposes `designations: ContributionDesignationSet` (never a single primary object), every line carries amount/currency/fund/fund-type/memo, reconciliation is explicit (`reconcilesToGiftAmount` + `issues[]`), no primary marker exists anywhere, and CRM/Hub summaries share `summarizeContributionDesignationSet` through the AL-256 loader. This PR closes the remaining acceptance gaps with two detail-level tests; no product code changes were needed.
- Test-only: +199 lines in `tests/unit/packages/api/admin/contribution-operations-read-model.test.ts`.

## Issue

Closes #259

## Scope

- In scope: detail-level acceptance tests for adjustment-driven designation rebuilds and fundless-line issue passthrough.
- Out of scope: adjustment record workflow itself (#260), approval policy (#261+).

## Acceptance criteria evidence

| Criterion | Implementation | Proof |
| --- | --- | --- |
| Detail exposes a designation set, not a primary object | `detail-read-model.ts:185` (`designations: ContributionDesignationSet`); payload has no `designation` property | existing split-gift test asserts `not.toHaveProperty("designation")` |
| Lines carry amount, currency, fund, fund type, memo/restriction | `contribution-shared.ts:68-80` line type; `designation-set.ts` builder | existing set + detail tests |
| Set reconciles to the effective gift amount | `designation-set.ts:177-182` | **New:** applied allocation-replacement adjustment rebuilds lines and reconciles to the effective 35,000 (not the original 30,000) |
| No designation treated as primary | no primary flag; split summaries render "N designations" | existing tests |
| CRM and Hub summaries derive from the same set | `summarizeContributionDesignationSet` via shared loader (AL-256) | shared row-contract tests + new detail summary assertion |
| Tests: split gifts, General Fund fallback, invalid fundless states | `contribution-designation-set.test.ts` + read-model tests | **New:** detail-level fundless line defaults to General Fund with the issue surfaced in the payload |

**Note on `restriction`:** the line type carries `restriction`, but no restriction data exists anywhere in the schema (`staged_gift_allocations` has no such column), so AC-2's "where available" is satisfied vacuously. Adding a schema column that nothing writes would be speculative scope; if restriction capture becomes a product requirement it needs its own issue defining the write path.

## Data/API Changes

None.

## UI Changes

None (the detail sheet already renders full sets, reconciliation warnings, and memos/restrictions when present).

## Testing

- ✅ `bunx vitest run tests/unit/packages/api/admin/contribution-operations-read-model.test.ts` — 17/17 (2 new)
- ✅ Full pre-push `ci:preflight` passed on push

## Follow-ups

- Next issue: #260 (adjustment records) — gap analysis complete; implementation already ships, PR will add idempotency edge-case tests.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR adds detail-level unit coverage for contribution designation truth. The main changes are:

- Acceptance coverage for rebuilding designation sets from applied allocation adjustments.
- Assertions that rebuilt designation lines reconcile against the effective gift amount.
- Coverage for fundless designation lines defaulting to General Fund while surfacing the payload issue.

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

This test-only PR is safe to merge with minimal risk.

The diff is limited to deterministic unit tests around an existing read-model builder. It follows the repository's Vitest testing guidance and introduces no runtime, schema, API, auth, or deployment changes.

No files require special attention.
</details>

<details><summary><h3><a href="https://www.greptile.com/trex">T-Rex</a> T-Rex Logs</h3></summary>

**What T-Rex did**
- Compared the Before and After contract states and found that the Before state included a General Fund fallback line with reconcilesToGiftAmount: true and a donor-intent note, and the After state produced the same contract fields with exit code 0.
- Validated that the after-run state matches the final contract fields with effective.amountCents: 35000, two designation lines totaling 35000, reconcilesToGiftAmount: true, no issues, sharedDesignationSummary.lineCount: 2, fundName: "2 designations", and hasTopLevelDesignation: false, with head exiting 0 and CONTRACT\_MATCH=true.

<a href="https://app.greptile.com/trex/runs/13752596/artifacts"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifactsDark.svg?v=4"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifacts.svg?v=4">View all artifacts</picture></a>

<sub><a href="https://www.greptile.com/trex">T-Rex</a> Ran code and verified through T-Rex</sub>
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| tests/unit/packages/api/admin/contribution-operations-read-model.test.ts | Adds two deterministic Vitest acceptance cases for contribution designation detail behavior; no material issues found. |

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
participant Test as Vitest detail acceptance tests
participant Builder as buildContributionDetail
participant Effective as deriveEffectiveContribution
participant Designations as buildContributionDesignationSet
participant Shared as buildSharedContributionRowFields

Test->>Builder: donation + allocations + adjustments
Builder->>Effective: original donation truth + applied adjustments
Effective-->>Builder: effective amount/fund/designation lines
Builder->>Designations: effective amount + designation lines/fund metadata
Designations-->>Builder: reconciled ContributionDesignationSet + issues
Builder->>Shared: designation set + effective row inputs
Shared-->>Builder: shared designation summary
Builder-->>Test: detail payload assertions
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
participant Test as Vitest detail acceptance tests
participant Builder as buildContributionDetail
participant Effective as deriveEffectiveContribution
participant Designations as buildContributionDesignationSet
participant Shared as buildSharedContributionRowFields

Test->>Builder: donation + allocations + adjustments
Builder->>Effective: original donation truth + applied adjustments
Effective-->>Builder: effective amount/fund/designation lines
Builder->>Designations: effective amount + designation lines/fund metadata
Designations-->>Builder: reconciled ContributionDesignationSet + issues
Builder->>Shared: designation set + effective row inputs
Shared-->>Builder: shared designation summary
Builder-->>Test: detail payload assertions
```

</a>
</details>

<sub>Reviews (6): Last reviewed commit: ["chore(pr-loop): sync PR with develop"](https://github.com/asymmetric-al/core/commit/1b9c5f60ae470b15882309cee69785cc4fa4fb17) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=41313309)</sub>

### CODE REVIEW

**Comment by II-ricky-bobby-II** [comment]
<sub>🤖 merge coordinator state (auto-updated): head `24bdb6705`, fix 0/3, update 0/3, nudges 2/2.</sub>

**Comment by II-ricky-bobby-II** [comment]
⚠️ Merge coordinator pausing automated handling: a Tier-1 review bot never posted on this head (nudges exhausted). Labeling `needs-human`. This is auto-recoverable — pushing a new commit (or the blocker clearing) un-parks it.

File: tests/unit/packages/api/admin/contribution-operations-read-model.test.ts [hunk 1]
```diff
@@ -472,6 +472,205 @@ describe("contribution operations detail read model", () => {
...
+        completedAt: "2026-05-20T00:00:00.000Z",
+        failedAt: null,
+        errorCode: null,
+        errorMessage: null,
+        refundedAt: null,
+        refundAmount: 0,
+        source: "online",
+        createdAt: "2026-05-20T00:00:00.000Z",
+        updatedAt: "2026-05-20T00:00:00.000Z",
+      },
```
[the remark this answers is not in the document]
Comment (cobmojo) [reply tests/unit/packages/api/admin/contribution-operations-read-model.test.ts:512]: Acknowledged — batching this test-fixture maintainability nit (Trivial / low-value) rather than chasing it individually, per the repo review guidelines. Not blocking; can fold into a later test-hygiene pass.

**Comment by cobmojo** [comment]
@greptile

**Comment by II-ricky-bobby-II** [comment]
✅ Merge coordinator resuming automated handling — the earlier blocker is no longer present.

**Comment by II-ricky-bobby-II** [comment]
⚠️ Merge coordinator pausing automated handling: a Tier-1 review bot never posted on this head (nudges exhausted). Labeling `needs-human`. This is auto-recoverable — pushing a new commit (or the blocker clearing) un-parks it.

**Review by II-ricky-bobby-II — COMMENTED** [review]
## pr-loop review (cycle 1)

Head: `24bdb6705940ac0ac889eee816e61cebe9110de1`
Findings: 0 reported, 0 confirmed blocker/high after adjudication.

**Review by II-ricky-bobby-II — COMMENTED** [review]
## pr-loop review (cycle 2)

Head: `1b9c5f60ae470b15882309cee69785cc4fa4fb17`
Findings: 0 reported, 0 confirmed blocker/high after adjudication.

File: tests/unit/packages/api/admin/contribution-operations-read-model.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (II-ricky-bobby-II) [reply tests/unit/packages/api/admin/contribution-operations-read-model.test.ts:512]: pr-loop: reviewed on head `1b9c5f60ae47` and adjudicated non-blocking (`informational`). Resolving to satisfy the all-conversations-resolved branch policy — re-open if you disagree.
